shotover-proxy/src/protocols/cassandra_protocol2.rs

```diff
@@ -584,45 +576,50 @@ impl Decoder for CassandraCodec2 {
         // if we need to close the connection return an error.
         if let Some(result) = self.force_close.take() {
             debug!("Closing errored connection: {:?}", &result);
-            return Err(anyhow::Error::msg(result));
+            return Err(anyhow!(result));
         }
-        debug!("Decoding {:?}", src.to_vec());
-        match self.decode_raw(src) {
-            Ok(Some(frame)) => {
-                debug!("Decoded {:?}", &frame);
+
+        match Frame::from_buffer(src, self.compressor) {
+            Ok((frame, bytes_read)) => {
+                // Clear the read bytes from the FramedReader
+                let _ = src.split_to(bytes_read);
+
                 Ok(Some(self.process_cassandra_frame(frame)))
             }
-            Ok(None) => Ok(None),
-            Err(e) => {
+            Err(ParseFrameError::NotEnoughBytes) => Ok(None),
```

Comment:
Does parsing the body length end up consuming bytes from the buffer or is it a copy operation?

Reply:
`Frame::from_buffer` can not consume bytes because it takes a data: `&[u8]`, its up to us to update the consumed bytes.